sympy/algebras/quaternion.py

```diff
@@ -300,7 +300,46 @@ def mul(self, other):
 
     @staticmethod
     def _generic_mul(q1, q2):
+        """Generic multiplication.
+
+        Parameters
+        ==========
+
+        q1 : Quaternion or symbol
+        q2 : Quaternion or symbol
+
+        It's important to note that if both q1 and q2 is not a Quaternion,
```

Comment:
It's important to note that if neither q1 nor q2 is a Quaternion,
        this function simply returns q1 * q2.